clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -663,7 +686,8 @@ invariant_checker_t::check_for_pc_discontinuity(per_shard_t *shard,
                   TRACE_MARKER_TYPE_KERNEL_XFER)) ||
             // We expect a gap on a window transition.
             shard->window_transition_ ||
-            shard->prev_instr_.instr.type == TRACE_TYPE_INSTR_SYSENTER;
+            shard->prev_instr_.instr.type == TRACE_TYPE_INSTR_SYSENTER ||
+            saw_repeated_syscall_instrs_with_same_pc;
```

Comment:
Having repeated syscalls implies an invalid non-branch flow. So it is confusing that valid_nonbranch_flow is set to true when saw_repeated_syscall_instrs_with_same_pc is true.

Reply:
After moving the duplicate syscall check inside the if-block, there is no need for this var. Removed it.